MREUnityRuntime/MREUnityRuntimeLib/Messaging/Commands/Command.cs

```diff
@@ -1,23 +1,26 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+using System;
 using System.Reflection;
 
 namespace MixedRealityExtension.Messaging.Commands
 {
     internal class Command<T> : ICommand where T : ICommandPayload
     {
         private readonly T _commandPayload;
+        private Action _onCompleteCallback;
 
-        public Command(T commandPayload)
+        public Command(T commandPayload, Action onCompleteCallback)
         {
             _commandPayload = commandPayload;
+            _onCompleteCallback = onCompleteCallback;
         }
 
         public void Execute(ICommandHandlerContext handlerContext, MethodInfo handlerMethod)
```

Comment:
>void [](start = 15, length = 4)

Question: Instead of an onCompleteCallback, what if this returned a Task that could be awaited?

Reply:
Is it necessary to have this wrapped in to a task rather than using the mechanism you have already built?  There is overhead associated with async methods and Tasks.  I would say that for the use case here, we should probably not be using tasks as the overhead is unnecessary.